taglib/CMakeLists.txt

```diff
@@ -364,8 +382,17 @@ set_target_properties(tag PROPERTIES
   LINK_INTERFACE_LIBRARIES ""
   PUBLIC_HEADER "${tag_HDRS}"
 )
+
 if(BUILD_FRAMEWORK)
-  set_target_properties(tag PROPERTIES FRAMEWORK TRUE)
+  set(BUILD_SHARED_LIBS On)
+  set(CMAKE_MACOSX_RPATH 1)
+  unset(INSTALL_NAME_DIR)
+  set_target_properties(tag PROPERTIES 
+      FRAMEWORK TRUE
+      MACOSX_RPATH 1
+      VERSION "A"
+      SOVERSION "A"
+      )
 endif()
```

Comment:
Personally, I believe the BUILD_FRAMEWORK changes should be in a separate pull request.